### PR TITLE
Add router factories

### PR DESCRIPTION
**Pull request by geospatial-jeff**
The goal of this PR is to restructure the library to be more similar to titiler, mostly by adding router factories.  This should give us a good starting point to pull in some of the lessons learned with titiler.  Trying my best to change the code as little as possible in doing so.

I'm avoiding base classes for the tiler/factory because I think that its too early to write a good one, especially when you consider that the base class should probably be data store agnostic.

### CODE REVIEW

File: timvt/endpoints/index.py [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-
-
-@router.get(
-    "/index",
-    response_model=List[TableMetadata],
-    tags=["Database"],
-    description="Return available tables.",
-    deprecated=True,
-)
-@router.get(
```
Comment (geospatial-jeff) [inline timvt/endpoints/index.py:19]: Removed this method entirely, it conflicts with the route to request a tilejson for a specific table - `/{table}.json`

File: timvt/endpoints/factory.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        @self.router.get("/tiles/{table}/{z}/{x}/{y}.pbf", **TILE_RESPONSE_PARAMS)
+        @self.router.get(
+            "/tiles/{TileMatrixSetId}/{table}/{z}/{x}/{y}.pbf", **TILE_RESPONSE_PARAMS
+        )
+        async def tile(
+            z: int = Path(..., ge=0, le=30, description="Mercator tiles's zoom level"),
+            x: int = Path(..., description="Mercator tiles's column"),
+            y: int = Path(..., description="Mercator tiles's row"),
+            tms: TileMatrixSet = Depends(self.tms_dependency),
+            table: TableParams = Depends(self.table_dependency),
```
Comment (geospatial-jeff) [inline timvt/endpoints/factory.py:104]: leaving this here for now, but I'd argue that any sort of table or layer configuration should be pushed lower into the `VectorTileReader`.  This keeps layers decoupled from the API layer.

File: timvt/endpoints/factory.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            """Return vector tile."""
+            timings = []
+            headers: Dict[str, str] = {}
+
+            reader = self.reader(db_pool, table=table, tms=tms)
+            with Timer() as t:
+                content = await reader.tile(x, y, z, columns=columns)
+            timings.append(("db-read", t.elapsed))
+
+            if timings:
```
Comment (vincentsarago) [inline timvt/endpoints/factory.py:116]: we can remove the `if timings` here!

**Review by vincentsarago — APPROVED** [review]
❤️

**Pull request merged**